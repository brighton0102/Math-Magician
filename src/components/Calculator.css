* {
  box-sizing: border-box;
  background-color: #505050;
  margin: 0;
  padding: 0;
}

.App h1 {
  text-align: center;
  margin-bottom: 10px;
}

.container {
  margin: auto;
  display: grid;
  max-width: 500px;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 1px;
  padding: 1px;
  background-color: #949494;
}

.btn-grid {
  grid-column: span 4;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 1px;
}

.btn-item {
  padding: 10px;
  font-size: 24px;
  text-align: center;
  background-color: #cac9c9;
  color: #fff;
  border: none;
  cursor: pointer;
}

.double {
  grid-column: span 2;
}

.display {
  grid-column: span 4;
  background-color: #929191;
  padding: 10px;
  font-size: 32px;
  text-align: right;
  color: #fff;
}

.btn-item:nth-child(4n),
.btn-item.equal {
  background-color: orange;
}
