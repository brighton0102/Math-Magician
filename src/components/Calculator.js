import './Calculator.css';
import React, { useState } from 'react';
import Button from './Button';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calculatorState, setCalculatorState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newState = calculate(calculatorState, buttonName);
    setCalculatorState(newState);
  };

  return (
    <section className="container">
      <div className="display">{calculatorState.next || calculatorState.total || '0'}</div>
      <div className="btn-grid">
        <Button label="AC" onClick={() => handleButtonClick('AC')} />
        <Button label="+/-" onClick={() => handleButtonClick('+/-')} />
        <Button label="%" onClick={() => handleButtonClick('%')} />
        <Button label="รท" onClick={() => handleButtonClick('รท')} />
        <Button label="7" onClick={() => handleButtonClick('7')} />
        <Button label="8" onClick={() => handleButtonClick('8')} />
        <Button label="9" onClick={() => handleButtonClick('9')} />
        <Button label="x" onClick={() => handleButtonClick('x')} />
        <Button label="4" onClick={() => handleButtonClick('4')} />
        <Button label="5" onClick={() => handleButtonClick('5')} />
        <Button label="6" onClick={() => handleButtonClick('6')} />
        <Button label="-" onClick={() => handleButtonClick('-')} />
        <Button label="1" onClick={() => handleButtonClick('1')} />
        <Button label="2" onClick={() => handleButtonClick('2')} />
        <Button label="3" onClick={() => handleButtonClick('3')} />
        <Button label="+" onClick={() => handleButtonClick('+')} />
        <Button label="0" type="double" onClick={() => handleButtonClick('0')} />
        <Button label="." onClick={() => handleButtonClick('.')} />
        <Button label="=" type="equal" onClick={() => handleButtonClick('=')} />
      </div>
    </section>
  );
};

export default Calculator;
