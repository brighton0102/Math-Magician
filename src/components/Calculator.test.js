import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

test('Calculator snapshot test', () => {
  const component = renderer.create(<Calculator />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
