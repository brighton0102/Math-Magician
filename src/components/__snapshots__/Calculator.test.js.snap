// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator snapshot test 1`] = `
<div
  className="calculator-container"
>
  <p
    className="math-description"
  >
    Let's do some Math!
  </p>
  <section
    className="container"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="btn-grid"
    >
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn-item double"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn-item undefined"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn-item equal"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </section>
</div>
`;
